pkg/agent/endpoints/endpoints.go

```diff
@@ -147,3 +152,8 @@ func (e *Endpoints) triggerListeningHook() {
 		e.hooks.listening <- struct{}{}
 	}
 }
+
+func (e *Endpoints) WaitForListening(listening chan struct{}) {
+	<-e.hooks.listening
+	listening <- struct{}{}
+}
```

Comment:
Similarly to the server side, should this check for non-nil of `e.hooks.listening`? I don't think it's worth checking fro `listening` to be non-nil.